boa/src/object/internal_methods.rs

```diff
@@ -881,6 +881,67 @@ impl GcObject {
         Ok(list)
     }
 
+    /// It is used to iterate over names of object's keys.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-enumerableownpropertynames
+    pub(crate) fn enumerable_own_property_names(
+        &self,
+        kind: PropertyNameKind,
+        context: &mut Context,
+    ) -> Result<Vec<JsValue>> {
+        // 1. Assert: Type(O) is Object.
+        // 2. Let ownKeys be ? O.[[OwnPropertyKeys]]().
+        let own_keys = self.own_property_keys();
+        // 3. Let properties be a new empty List.
+        let mut properties = vec![];
+
+        // 4. For each element key of ownKeys, do
+        for key in own_keys {
+            // a. If Type(key) is String, then
+            if let PropertyKey::String(key_str) = &key {
+                // i. Let desc be ? O.[[GetOwnProperty]](key).
+                let desc = self.__get_own_property__(&key);
+                // ii. If desc is not undefined and desc.[[Enumerable]] is true, then
+                if let Some(desc) = desc {
+                    if desc.expect_enumerable() {
+                        // 1. If kind is key, append key to properties.
+                        if let PropertyNameKind::Key = kind {
+                            properties.push(JsValue::String(key_str.clone()))
```

Comment:
```suggestion
                            properties.push(key_str.clone().into());

```
We can use `into` without losing readability because we know `key_str` is a string by its name.